instrumentation/dalli/lib/opentelemetry/instrumentation/dalli/patches/server.rb

```diff
@@ -25,6 +25,7 @@ def request(op, *args) # rubocop:disable Naming/MethodParameterName
             end
 
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
+            attributes.compact!
```

Comment:
We typically want to avoid using `compact!` whenever possilbe since it is a bit slower than conditional assigments. 

You can see a benchmark here someone shared:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/693#discussion_r1360669417